tests/integration_tests/functional/test_balloon.py

```diff
@@ -298,6 +298,16 @@ def test_reinflate_balloon(test_microvm_with_ssh_and_balloon, network_config):
     firecracker_pid = test_microvm.jailer_clone_pid
     ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
 
+    # First inflate the balloon to free up the uncertain amount of memory
+    # used by the kernel at boot and establish a baseline, then give back
+    # the memory.
+    response = test_microvm.balloon.patch(amount_mb=200)
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+    time.sleep(5)
```

Comment:
Sleeping for an amount of time doesn't really guarantee that this test won't fail anymore. It also adds extra time to the test suite.

Could we instead check in a loop that the requested amount has been freed? If we exceed a large enough timeout we can fail.

The key is that instead of relying on a fixed amount of time to hopefully free the requested amount of memory, we sometimes may free it quicker than the 5 seconds and some other times need 6 seconds. If it doesn't happen within a reasonable time frame, then something is wrong and we would need to fail.

Reply:
The sleep time was one thing that previously caused intermittent failures. Indeed, if the sleep time was too low it wouldn't have time to free the memory, but 5 seconds for 200 MiB is sufficient in all cases. The real fix is establishing the baseline before running the test by constricting the VM to 56 MiB,  releasing the memory, and then start actually measuring the RSS.

I agree that the sleep usage is not ideal, but this is not the only place where it's used. To get rid of the sleeps, we need to perform the checks via a polling mechanism (similar to what you described) and that is a separate issue in my opinion.

Reply:
> that is a separate issue in my opinion

I don't feel strongly towards doing the fix in one change or two, but it sounds like we're introducing something now which we will remove later in order to fix the same underlying problem.

Reply:
Again, all the balloon tests currently use `sleep` to wait for inflate/deflate completion. We should indeed introduce a mechanism more efficient than the `sleep` one in the future and it should replace the `sleep`s in the other tests as well, but for the purpose of fixing the intermittent failure, this fix doesn't introduce anything more than an extra inflate/deflate sequence.

Reply:
At least open an issue to address all of these `sleep`s. We can pick it up later, it'd also be a good ramp-up issue.

Reply:
Created the related issue separately.

#2261

Reply:
Let's merge this for now as is. As per the discussion in #2257, the question that I rased above will be picked up by someone else.